torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -377,9 +386,10 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
               inner_reduction_unroll_factor * outer_reduction_unroll_factor *
               4 >
           255 * 3 &&
-      bdimx * bdimy * bdimz * 2 <= device_max_threads_per_multiprocessor &&
-      batches_per_block_inner_reduction >= 2) {
-    batches_per_block_inner_reduction /= 2;
+      bdimx * bdimy * bdimz * 2 <= max_threads_in_block &&
+      batches_per_block_inner_reduction >=
+          batches_per_block_inner_reduction_max) {
+    batches_per_block_inner_reduction = batches_per_block_inner_reduction / 2;
```

Comment:
Here, it seems we are trying to decrease `batches_per_block_inner_reduction` until hitting some threshold. The threshold was 2 before, but now it's `batches_per_block_inner_reduction_max`, which is 10, so that `batches_per_block_inner_reduction` is at least 5. Is it correct? If so, it seems `batches_per_block_inner_reduction_max` actually gives a minimum for the buffer size, doesn't it?

Reply:
Yes. we want to make it smaller than `batches_per_block_inner_reduction_max` so each thread won't use too many registers and doing too many serial works. There is no minmum value for `batches_per_block_inner_reduction`, it will be reduced if it is too large and bdimx is not exceeding max threads. so it only has a maximum.